README.md

```diff
@@ -1,30 +1,80 @@
 # Esy
 
-`package.json` workflow for compiled languages.
+`package.json` workflow for native development with Reason/OCaml.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [What](#what)
+  - [For npm users](#for-npm-users)
+  - [For OPAM users](#for-opam-users)
+  - [In depth](#in-depth)
+- [Install](#install)
+- [Workflow](#workflow)
+  - [Try An Example](#try-an-example)
+  - [Configuring Your `package.json`](#configuring-your-packagejson)
+    - [Specify Build & Install Commands](#specify-build--install-commands)
+      - [`esy.build`](#esybuild)
+      - [`esy.install`](#esyinstall)
+    - [Enforcing Out Of Source Builds](#enforcing-out-of-source-builds)
+    - [Exported Environment](#exported-environment)
+  - [Esy Environment Reference](#esy-environment-reference)
+    - [Build Environment](#build-environment)
+    - [Command Environment](#command-environment)
+  - [Esy Command Reference](#esy-command-reference)
+- [How Esy Works](#how-esy-works)
+  - [Build Steps](#build-steps)
+  - [Directory Layout](#directory-layout)
+    - [Global Cache](#global-cache)
+    - [Top Level Project Build Artifacts](#top-level-project-build-artifacts)
+- [Developing](#developing)
+  - [Running Tests](#running-tests)
+  - [Issues](#issues)
+  - [Publishing Releases](#publishing-releases)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
 ## What
 
-- Makes compiled projects work with `package.json` package managers.
+### For npm users
 
-- Makes it trivial to share natively compiled projects with anyone - even if they don't
-  have a package manager installed.
+For those familiar with [npm][], esy allows to work with Reason/OCaml projects
+within the familiar npm-like workflow:
 
-- Makes native projects build faster.
+- Declare dependencies in `package.json`.
 
-- Widens the reach of OPAM packages by allowing them to be installed
-  by `npm` (the most popular package manager).
+- Install and build with `esy install` and `esy build` commands. Dependencies'
+  source code end up in `node_modules`.
 
-## Features
+- Share your work with other developers by publishing on npm registry and/or github.
 
-- Directory based projects (like `package.json`).
+- Access packages published on [OPAM][] (a package registry for OCaml) via
+  `@opam` npm scope (for example `@opam/lwt` to pull `lwt` library from OPAM).
+
+### For OPAM users
+
+For those who familiar with [OPAM][], esy provides a powerful alternative:
+
+- Manages OCaml compilers and dependencies on a per project basis.
+
+- Sandboxes project environment by exposing only those packages which are
+  defined as dependencies.
+
+- Fast parallel builds which are agressively cached.
```

Comment:
I think the ability for the cache to be shared across projects is a killer feature worth mentioning. It's saved me literal hours by now.